2020-01-Winter/1_linked_lists/README.md

```diff
@@ -140,19 +141,76 @@ Go to [Top](#top)
 
 ### 3. SOLUTION 3 TODO :bug:
 
-Source: TODO :bug:
+Source: Cracking the Code Interview Book, Chapter 2, Question 2.1
 
-#### Naive/Simple Solution 
+#### Solution 
 
-TODO :bug:
+With the Buffer: Use a set to keep track of what integers have been encountered.
 
-#### Optimal Solution
+```
+void RemoveDuplicatesWithBuffer(Node *Start) {
+  if (Start == nullptr || Start->GetNext() == nullptr) {
+    return;
+  }
 
-TODO :bug:
+  unordered_set<int> set;
+
+  Node *Curr = Start;
+  Node *Prev = nullptr;
+
+  while (Curr != nullptr) {
+    // Doesn't exist
+    if (set.find(Curr->GetValue()) == set.end()) {
+
+      set.insert(Curr->GetValue());
+      Prev = Curr;
+      Curr = Curr->GetNext();
+
+    } else { // Exists
+
+      Prev->SetNext(Curr->GetNext());
+      delete Curr;
+      Curr = Prev->GetNext();
+    }
+  }
+}
+```
+
+Without a Buffer: Have a pointer/runner that iterates over the list to check for duplicates.
```

Comment:
Nitpick: Again, short and sweet, which is good. It might be prudent to specify that time complexity increases with this approach, which is a trade-off for eliminating extra memory allocation. If you search for duplicates on each element but there are no duplicates, you have worst case complexity of `O(n^2)`, `n` being the number of traversal operations.